blocks/algolia-assortment-content-source-block/sources/algolia-assortment.js

```diff
@@ -0,0 +1,49 @@
+import algoliasearch from "algoliasearch/lite";
+import { algoliaSearchKey, algoliaAppId } from "fusion:environment";
+
+const params = {
+	filters: "text",
+	index: "text",
+	query: "text",
+	ruleContexts: "text",
+	hitsPerPage: "number",
+	page: "number",
+};
+
+const fetch = async (contentSourceParameters) => {
+	const { filters, index, query, ruleContexts } = contentSourceParameters;
+	let { hitsPerPage, page } = contentSourceParameters;
+
+	// parameters in content source are defined as null by default
+	if (hitsPerPage === null) {
+		hitsPerPage = 20;
+	}
+	if (page === null) {
+		page = 0;
+	}
+	const client = algoliasearch(algoliaAppId, algoliaSearchKey);
+
+	const clientSearchIndex = client.initIndex(index);
+	try {
+		const { hits } = await clientSearchIndex.search(query, {
+			// array of strings
+			// https://www.algolia.com/doc/api-reference/api-parameters/ruleContexts/
+			ruleContexts: [ruleContexts],
+			filters,
+			hitsPerPage,
+			page,
+		});
+		return hits;
+	} catch (err) {
+		const error = new Error(err);
+		error.statusCode = 500;
+		throw error;
+	}
+};
+export default {
+	fetch,
+	params,
+	transform: (data) => data,
+	// Time to live cache set to the Fusion minimum of 2 minutes (120 seconds)
+	ttl: 120,
```

Comment:
@JackHowa : FYI as a topic to revisit but not a blocker. The Commerce team has mentioned being amenable to a 5-minute TTL for assortment data. Delivery may appreciate that as well.

Reply:
gotcha. I was going off of the ticket requirements. if that's not the goal, then I can change this @ecarlisle

Reply:
@ecarlisle Where was this mentioned? As we refined the ticket with the 2 minute ttl - I want to make sure we are all on the same page with decisions being made.
We can move forward with the 2 minute ttl for now, and create a ticket to update it to something else if needed